h3i/src/prompts/h3/mod.rs

```diff
@@ -423,6 +429,23 @@ fn prompt_extension() -> InquireResult<Action> {
     Ok(action)
 }
 
+pub fn prompt_connection_close() -> InquireResult<Action> {
+    let is_app = prompt_yes_no("application close:")?;
+    let error_code = prompt_varint("error code:")?;
+    let reason = Text::new("reason:")
+        .with_placeholder("empty")
+        .prompt()
+        .unwrap_or(String::new());
+
+    Ok(Action::ConnectionClose {
+        error: ConnectionError {
+            is_app,
+            error_code,
+            reason: reason.as_bytes().to_vec(),
+        },
+    })
```

Comment:
Once a connection is closed, many other actions are no longer possible. It would be nice to find a way for the prompter to reflect that to the user. This might require customizing the top-level suggestions returned in the loop in `prompt()`

Reply:
After messing around with this, I think it's a larger change that warrants a separate PR. `prompt_action` will need to know about connection state at that point

Reply:
all good